addons/hr_timesheet/migrations/10.0.1.0/post-migration.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+def migrate_allow_timesheets(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE project_project pp
+        SET allow_timesheets = True
+        FROM account_analytic_account aaa
+        WHERE aaa.id = pp.analytic_account_id
+        AND aaa.%s
+        """ % openupgrade.get_legacy_name('use_timesheets'),
+    )
+
+
+def fill_analytic_line_project(env):
+    """Fill project with the linked one in the related task or issue."""
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_analytic_line aal
+        SET project_id = pt.project_id
+        FROM project_task pt
+        WHERE pt.id = aal.task_id
+        AND pt.project_id IS NOT NULL
+        AND aal.project_id IS NULL
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_analytic_line aal
+        SET project_id = pi.project_id
+        FROM project_issue pi
+        WHERE pi.id = aal.issue_id
+        AND pi.project_id IS NOT NULL
+        AND aal.project_id IS NULL
+        """,
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    migrate_allow_timesheets(env)
+    fill_analytic_line_project(env)
```

Comment:
In my opinion we need to consider that in v9 the analytic lines were linked to analytic account. 
The basic migration process should consist of detecting the analytic lines with `is_timesheet = True`, then look for an existing project associated to the analytic account of the line, and use it. If no project exists, create one.

Because there are companies that used analytic accounts not connected to projects to report time (e.g. Office time), and in v10 this is no longer allowed. But we need to respect the time entered in the past on those analytic accounts.

This is very, very important.